### PR TITLE
ICU-21394 Don't use null to test for unset ICU4J NumberFormatter options

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21394
- [x] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [x] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/test/intltest/numbertest_api.cpp [hunk 1]
```diff
@@ -798,6 +798,15 @@ void NumberFormatterApiTest::unitMeasure() {
...
             u"-1 Std., 14 Min. und 24 Sek.");
+
+    assertFormatSingle(
+            u"Zero out the unit field",
+            u"",
+            u"",
+            NumberFormatter::with().unit(KELVIN).unit(MeasureUnit()),
+            Locale("en"),
+            100,
+            "100");
```
Comment (markusicu) [inline icu4c/source/test/intltest/numbertest_api.cpp:809]: optional: prefix u here as well

**Review by markusicu — DISMISSED** [review]
lgtm
good fix

File: icu4c/source/test/intltest/numbertest_api.cpp (hunk 1 above)
Comment (sffc) [reply icu4c/source/test/intltest/numbertest_api.cpp]: Fixed in 151c55a and then squashed

**Comment by sffc** [comment]
I guess I should make the same fix in NumberRangeFormatterSettings.java.  I will follow up

**Review by hugovdm — APPROVED** [review]
Neat!

**Review by markusicu — APPROVED** [review]
lgtm

**Comment by sffc** [comment]
The Travis error is:

```
[ivy:retrieve] 		::::::::::::::::::::::::::::::::::::::::::::::
[ivy:retrieve] 		::              FAILED DOWNLOADS            ::
[ivy:retrieve] 		:: ^ see resolution messages for details  ^ ::
[ivy:retrieve] 		::::::::::::::::::::::::::::::::::::::::::::::
[ivy:retrieve] 		:: junit#junit;4.12!junit.jar(javadoc)
[ivy:retrieve] 		::::::::::::::::::::::::::::::::::::::::::::::
```

I'm going to override and merge the PR.

**Pull request merged**